src/service/command/execute-command-service.ts

```diff
@@ -22,14 +22,14 @@ export class ExecuteCommandService {
     return this._commandExecutorDelegator.executeCommand(treatedCommand, cwd);
   }
 
-  public async executeChainCommands(nodes: Node[], executionPhase: ExecutionPhase, cwd?: string): Promise<ExecuteNodeResult[]> {
+  public async executeChainCommands(nodes: NodeExecution[], executionPhase: ExecutionPhase): Promise<ExecuteNodeResult[]> {
     const result: ExecuteNodeResult[] = [];
     for (const node of nodes.values()) {
```

Comment:
with `const {node, cwd} of nodes` it is less verbose, wdyt?
why `nodes.values()`? Am I missing anything here? otherwise `for (const {node, cwd} of nodes) {`

```suggestion
        for (const {node, cwd} of nodes) {
            const commands = this.getNodeCommands(node, executionPhase, this._configurationService.getNodeExecutionLevel(node));
            if (commands?.length) {
                result.push(await this.executeNodeCommands(node, commands, this._configurationService.skipExecution(node), cwd));
            } else {
                result.push({node});
            }
        }
```

this would be a clear `reduce` example, just the `await` is not allowing it....

Reply:
Oh I did not change the `for` loop structure. This is how it was originally :sweat_smile: 
But yes I agree `const {node, cwd} of nodes` is a better choice